Formula/pop.rb

```diff
@@ -0,0 +1,23 @@
+class Pop < Formula
+  desc "Send emails from your terminal"
+  homepage "https://github.com/charmbracelet/pop"
+  url "https://github.com/charmbracelet/pop/archive/v0.1.0.tar.gz"
+  sha256 "d5cedd18327131df4e3857ddbc0d98326cab619630eb89f7ce2f2d03a3f4e723"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X main.Version=#{version}")
+
+    man_page = Utils.safe_popen_read(bin/"pop", "man")
+    (man1/"pop.1").write man_page
+
+    generate_completions_from_executable(bin/"pop", "completion")
+  end
+
+  test do
+    assert_match "pop version", shell_output("#{bin}/pop --version").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.